porch/pkg/git/git.go

```diff
@@ -927,7 +961,12 @@ func (r *gitRepository) loadTasks(ctx context.Context, startCommit *object.Commi
 		}
 
 		for _, gitAnnotation := range gitAnnotations {
-			if gitAnnotation.PackagePath == packagePath && gitAnnotation.Revision == revision {
+			packageMatches := gitAnnotation.PackagePath == packagePath
+			descriptionMatches := gitAnnotation.Description == description ||
+				// this is needed for porch package revisions created before the description field existed
+				(gitAnnotation.Revision == description && gitAnnotation.Description == "")
+
+			if packageMatches && descriptionMatches {
```

Comment:
Here is the scenario in my mind:

I create a porch package draft (before the description field was added). I give it a revision of v1. I do some work on it, so that it has some commits with the annotation revision:v1.

This PR goes in. My porch package draft now has an empty revision but a description v1. I do some work so this package revision now has some commits with annotation description:v1. When I load up tasks, I need to look for:
- commits that have a revision:v1, but no description.
- commits with a description:v1. 

Then let's say I publish the package revision and it gets auto-assigned a revision of v3. When I load tasks, what I look for is unchanged:
- commits that have a revision:v1, but no description.
- commits with a description:v1. 

For porch packages revisions that were created after the description field was introduced, all git commit annotations should have a populated description field, so the check here is still valid (just match based on the description).